ansible_events/key.py

```diff
@@ -0,0 +1,39 @@
+import asyncio
+import logging
+import os
+import shutil
+import stat
+import tempfile
+
+logger = logging.getLogger()
+
+ssh_add = shutil.which("ssh-add")
+
+
+async def install_private_key(private_key):
+
+    local_working_directory = tempfile.mkdtemp()
+    try:
+
+        key_file_fd = os.open(
+            os.path.join(local_working_directory, "key"),
+            os.O_WRONLY | os.O_CREAT | os.O_EXCL,
+            mode=stat.S_IRUSR | stat.S_IWUSR,
```

Comment:
A comment explaining combination of flags would be nice here. Something like:

`# Create a file for write with mode 0600`